src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskIntegrationTest.java

```diff
@@ -566,8 +580,20 @@ void testCopyingExistingWithARestartMidwayThrough() {
 
       List<SourceRecord> thirdPoll = getNextBatch(task);
       assertSourceRecordValues(createInserts(26, 50), thirdPoll, coll);
+      // Make sure all elements, except the last one, contains the "copy" key
+      assertTrue(
+          thirdPoll.stream()
+              .map(SourceRecord::sourceOffset)
+              .limit(24)
+              .allMatch(i -> i.containsKey("copy")));
+      // Make sure the last copied element does not contain the "copy" key
       assertTrue(
-          thirdPoll.stream().map(SourceRecord::sourceOffset).allMatch(i -> i.containsKey("copy")));
+          thirdPoll.stream()
+              .map(SourceRecord::sourceOffset)
+              .skip(24)
+              .findFirst()
+              .filter(i -> !i.containsKey("copy"))
+              .isPresent());
 
```

Comment:
Can we add a test where a restart happens (similar to how it's done [here](https://github.com/mongodb/mongo-kafka/blob/1a09960e1f90c7ab077c1f1d6a502d162f88e5fb/src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskIntegrationTest.java#L574-L575)) after copying all documents and confirming that a second poll won't fetch any data?

Reply:
Hmm its sort of misleading but that test you linked technically doesn't correctly test restarts because these tests aren't referencing a cached state.. i.e. shouldCopyData is always true in this case because `context` which is a magic value that comes from Kafka is always null.

Let me check to make sure this isn't something that is wrong with my environment, potentially I'm missing a setup step.

Reply:
Ok I can actually mock this